cassandra/__init__.py

```diff
@@ -728,3 +728,32 @@ class UnresolvableContactPoints(DriverException):
     contact points, only when lookup fails for all hosts
     """
     pass
+
+
+class OperationType:
+        Read = 0
+        Write = 1
+
+        def __init__(self, value):
+            if value == self.Read:
+                self.type = "Read"
+            elif value == self.Write:
+                self.type = "Write"
+            else:
+                self.type = "Other"
+
+        def __str__(self):
+            return self.type
+
+class RateLimitReached(ConfigurationException):
+    '''
+    Rate limit was exceeded for a partition affected by the request.
+    '''
+    op_type = None
+    rejected_by_coordinator = False
+
+    def __init__(self, op_type=None, rejected_by_coordinator=False):
+        self.op_type = op_type
+        self.rejected_by_coordinator = rejected_by_coordinator
+        message = "[request_error_rate_limit_reached OpType=" + op_type.type + " RejectedByCoordinator=%t]"+ str(rejected_by_coordinator)
+        Exception.__init__(self, message)
```

Comment:
I'm not familiar with this part of the code. Why is it needed to have 2 exceptions ('RateLimitReached` and `RateLimitReachedException`)?

Reply:
I believe this is the convention, `RateLimitReahcedException` is a subclass of ErrorMessage and it has to have fields `opcode` and `summary`, but to have fields specific for the exception there is `RateLimitReached`, I was trying to be consisent with this convention (see `AlreadyExists` and `AlreadyExistsException`)